### PR TITLE
driver: adc: Use logging API

### PR DESCRIPTION
**Pull request by ene-steven**
Use logging API instead of printk
Remove ampersand for function

### CODE REVIEW

**Review by kartben — DISMISSED** [review]

**Comment by kartben** [comment]
ping @anangl

**Review by anangl — DISMISSED** [review]

**Review by gmarull — CHANGES_REQUESTED** [review]
please, improve commit title. this is just touching a single driver.

File: drivers/adc/adc_ene_kb1200.c [hunk 1]
```diff
@@ -11,10 +11,13 @@
 #include <zephyr/drivers/pinctrl.h>
 #include <errno.h>
 #include <reg/adc.h>
+#include <zephyr/logging/log.h>
 
 #define ADC_CONTEXT_USES_KERNEL_TIMER
 #include "adc_context.h"
 
+LOG_MODULE_REGISTER(adc, CONFIG_ADC_LOG_LEVEL);
```
Comment (gmarull) [inline drivers/adc/adc_ene_kb1200.c:19]:
```suggestion
LOG_MODULE_REGISTER(ene_kb1200, CONFIG_ADC_LOG_LEVEL);
```

Comment (nashif) [reply drivers/adc/adc_ene_kb1200.c:19]: and to be consistent with all other ADC drivers, it should be `LOG_MODULE_REGISTER(adc_ene_kb1200...`

**Review by gmarull — APPROVED** [review]

**Review by anangl — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Pull request merged**